### PR TITLE
Add `SecretManage::sign_evm`

### LINKED ISSUE #505: Add sign EVM to the SecretManage trait
Also expose to nodejs and python bindings

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This adds the `SecretManage::sign_evm` method, but not any other implementation, in order to minimize the changes in the PR. I will create more PRs on top of this to implement bindings and such.

NOTE: This PR is based on #453

## Links to any relevant issues

Fixes #505 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Needs to add bindings

File: sdk/src/client/secret/mnemonic.rs [hunk 1]
```diff
@@ -97,6 +97,19 @@ impl SecretManage for MnemonicSecretManager {
 
         Ok(Ed25519Signature::new(public_key, signature))
     }
+
+    async fn sign_evm(
+        &self,
+        msg: &[u8],
+        chain: &Chain,
+    ) -> Result<(secp256k1_ecdsa::PublicKey, secp256k1_ecdsa::Signature), Self::Error> {
+        // Get the private and public key for this Ed25519 address
```
Comment (Thoralf-M) [inline sdk/src/client/secret/mnemonic.rs]: Not Ed25519

**Comment by DaughterOfMars** [comment]
> Needs to add bindings

Nah, I'm gonna do that in another PR to avoid this one blowing up like the last one.

**Comment by thibault-martinez** [comment]
> > Needs to add bindings
> 
> Nah, I'm gonna do that in another PR to avoid this one blowing up like the last one.

Fair enough, but please prioritise this then, as Firefly is waiting on it.

File: sdk/src/client/secret/mod.rs [hunk 2]
```diff
@@ -78,9 +81,16 @@ pub trait SecretManage: Send + Sync {
         options: impl Into<Option<GenerateAddressOptions>> + Send,
     ) -> Result<Vec<EvmAddress>, Self::Error>;
 
-    /// Signs `msg` using the given [`Chain`].
+    /// Signs ed25519 key using the given [`Chain`].
     async fn sign_ed25519(&self, msg: &[u8], chain: &Chain) -> Result<Ed25519Signature, Self::Error>;
 
+    /// Signs evm key using the given [`Chain`].
+    async fn sign_evm(
```
Comment (thibault-martinez) [inline sdk/src/client/secret/mod.rs:88]: Changelog

File: sdk/src/client/secret/mod.rs [hunk 3]
```diff
@@ -78,9 +81,16 @@ pub trait SecretManage: Send + Sync {
         options: impl Into<Option<GenerateAddressOptions>> + Send,
     ) -> Result<Vec<EvmAddress>, Self::Error>;
 
-    /// Signs `msg` using the given [`Chain`].
+    /// Signs ed25519 key using the given [`Chain`].
```
Comment (thibault-martinez) [inline sdk/src/client/secret/mod.rs]: ?

Comment (DaughterOfMars) [reply sdk/src/client/secret/mod.rs]: ?

Comment (thibault-martinez) [reply sdk/src/client/secret/mod.rs]: That's a bit weird? We're not signing a key but a message?

Comment (DaughterOfMars) [reply sdk/src/client/secret/mod.rs]: Oh, gotcha

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**